third_party/blink/renderer/core/page/chrome_client_impl.cc

```diff
@@ -395,12 +395,26 @@ void ChromeClientImpl::AddMessageToConsole(LocalFrame* local_frame,
                                            unsigned line_number,
                                            const String& source_id,
                                            const String& stack_trace) {
+  // [RUN-2387] Make sure that stack_trace has consistent length, despite being
+  // slightly divergent in content.
+  unsigned stack_trace_length = (unsigned)recordreplay::RecordReplayValue(
+      "ChromeClientImpl::AddMessageToConsole stack_trace.length",
+      (uintptr_t)stack_trace.length());
+  const String* new_stack_trace_ptr;
+  String new_stack_trace;
+  if (stack_trace_length != stack_trace.length()) {
+    std::string stack_trace_str = stack_trace.Ascii();
+    stack_trace_str.resize(stack_trace_length, ' ');
+    new_stack_trace_ptr = &(new_stack_trace = String::FromUTF8(
+                                &stack_trace_str[0], stack_trace_length));
+  } else {
+    new_stack_trace_ptr = &stack_trace;
+  }
```

Comment:
It seems better if we record/replay the stack trace string and not just the length.  If anything tries to read this stack trace string later on they will get divergent contents.